mlrun/common/schemas/common.py

```diff
@@ -41,3 +43,44 @@ class ImageBuilder(pydantic.BaseModel):
 
     class Config:
         extra = pydantic.Extra.allow
+
+
+class LabelsModel(pydantic.BaseModel):
+    """
+    A model for handling labels.
+
+    This class accepts either a dictionary or a list for filtering by labels.
+
+    :param labels:
+            - If a dictionary is provided, it should be in the format
+              {'label_name': 'value'}. This will be converted to a list of strings
+              in the format 'label_name=value'.
+            - If a list is provided, all items must be strings. Each string can either
+              be a simple label name (e.g., 'label1') or a key-value pair in the format
+              'label=value'.
+            - If no labels are specified, the default is an empty list.
+    """
+
+    labels: typing.Optional[typing.Union[dict[str, str], list[str]]]
+
+    @pydantic.validator("labels")
+    @classmethod
+    def validate(cls, labels) -> list[str]:
+        if labels is None:
+            return []
+
+        if isinstance(labels, list):
+            if not all(isinstance(item, str) for item in labels):
+                raise mlrun.errors.MLRunValueError(
+                    "All items in the list must be strings."
+                )
+            return labels
+
+        if isinstance(labels, dict):
+            if not labels:
+                return []
+            return [f"{key}={value}" for key, value in labels.items()]
```

Comment:
We also support giving mix of labels with values and labels without. So a dict like:
```
{"a": "b", "a": None}
```
Should be valid and should output to
```
["a=b", "a"]
```